redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/NamedRedisClientFactory.java

```diff
@@ -79,7 +84,29 @@ public RedisClient redisClient(NamedRedisServersConfiguration config, @Nullable
     @EachBean(NamedRedisServersConfiguration.class)
     public StatefulRedisConnection<K, V> redisConnection(NamedRedisServersConfiguration config) {
         RedisCodec<K, V> namedCodec = beanLocator.findBean(RedisCodec.class, Qualifiers.byName(config.getName())).orElse(defaultCodec);
-        return super.redisConnection(getRedisClient(config), namedCodec);
+
+        StatefulRedisConnection<K, V> connection;
+        if (config.getUri().isPresent() && !config.getReplicaUris().isEmpty()) {
+            List<RedisURI> uris = new ArrayList<>(config.getReplicaUris());
+            uris.add(config.getUri().get());
+
+            connection = MasterReplica.connect(
+                getRedisClient(config),
+                namedCodec,
+                uris
+            );
+            if (config.getReadFrom() != null) {
+                ((StatefulRedisMasterReplicaConnection<K, V>) connection).setReadFrom(config.getReadFrom());
+            }
+        } else {
+            connection = super.redisConnection(getRedisClient(config), namedCodec);
+
+            if (connection instanceof StatefulRedisClusterConnection && config.getReadFrom() != null) {
```

Comment:
use `connection instanceof StatefulRedisClusterConnection<?, ?> conn` to avoid the cast